### PR TITLE
🔧 Redirect ampproject.org to amp.dev

### PR DESCRIPTION
**Pull request by matthiasrohmer**

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -24,1610 +24,648 @@
...
       },
       {
-        "source": "/:lang/docs/reference/validation_errors",
-        "destination": "/:lang/docs/troubleshooting/validation_errors.html",
+        "source": "/:lang/docs/interaction_dynamic/interactivity",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/develop/interactivity/create-interactive.html",
         "type": 301
       },
       {
-        "source": "/:lang/docs/tutorials/add
```
Comment (matthiasrohmer) [inline firebase.json]: Rereading the documentation this probably needs to be `**`, doesn't it? See: https://firebase.google.com/docs/hosting/full-config#redirects

File: firebase.json [hunk 2]
```diff
@@ -24,1610 +24,648 @@
...
-      {
-        "source": "/docs/tutorials/add_advanced/setting_up?(.html)",
-        "destination": "/docs/fundamentals/add_advanced/setting_up.html",
-        "type": 301
-      },
-      {
-        "source": "/docs/tutorials/add_advanced/tracking_data?(.html)",
-        "destination": "/docs/fundamentals/add_advanced/tracking_data.html",
+        "source": "/:lang/docs/ads",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/?format=ads?referrer=ampproject.org",
```
Comment (sebastianbenz) [inline firebase.json]: double `?`

File: firebase.json [hunk 3]
```diff
@@ -24,1610 +24,648 @@
...
-        "destination": "/docs/getting_started/visual_story/congratulations.html",
+        "source": "/:lang/docs/design/responsive_amp",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/develop/style_and_layout/?referrer=ampproject.org",
         "type": 301
       },
       {
-        "source": "/docs/design/visual_story/congratulations?(.html)",
-        "destination": "/docs/getting_started/visual_story/congratulations.html",
+        "source": "/:lang/docs/fundamentals/",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/?referrer=ampproject.org",
```
Comment (sebastianbenz) [inline firebase.json]: should this go to `start`?

File: firebase.json [hunk 4]
```diff
@@ -24,1610 +24,648 @@
...
-        "destination": "/:lang/docs/analytics/deep_dive_analytics.html",
+        "source": "/:lang/docs/getting_started/create/publish",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/start/create/publish?referrer=ampproject.org",
         "type": 301
       },
       {
-        "source": "/:lang/docs/ads_analytics/deep_dive_analytics",
-        "destination": "/:lang/docs/analytics/deep_dive_analytics.html",
+        "source": "/:lang/docs/getting_started/quickstart",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/?referrer=ampproject.org",
```
Comment (sebastianbenz) [inline firebase.json]: feels like there should be a better link for t his

File: firebase.json [hunk 5]
```diff
@@ -24,1610 +24,648 @@
...
-        "destination": "/:lang/docs/design/responsive/responsive_design.html",
+        "source": "/:lang/docs/integration/integrate-your-tech",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/contribute/integrate-your-tech?referrer=ampproject.org",
         "type": 301
       },
       {
-        "source": "/:lang/docs/guides/responsive/style_pages",
-        "destination": "/:lang/docs/design/responsive/style_pages.html",
+        "source": "/:lang/docs/integration/pwa-amp",
+        "destination": "https://amp.dev/:lang/documentation/guides-and-tutorials/integrate/amp-in-pw
```
Comment (sebastianbenz) [inline firebase.json]: this should go to `develop`

Comment (sebastianbenz) [inline firebase.json]: this should go to https://amp.dev/documentation/guides-and-tutorials/develop/media_iframes_3p/amp_replacements.html?format=websites

Comment (sebastianbenz) [inline firebase.json]: this should go to https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/how-cached.html?format=websites

Comment (matthiasrohmer) [reply firebase.json]: So resort the document from Integrate over to Develop?

File: firebase.json (hunk 4 above)
Comment (matthiasrohmer) [reply firebase.json]: What do you think of: https://amp.dev/documentation/guides-and-tutorials/start/create.html?format=websites?

File: firebase.json (hunk 3 above)
Comment (matthiasrohmer) [reply firebase.json]: We don't have bucket landing pages yet - I think Paul wasn't a fan of them: https://amp.dev/documentation/guides-and-tutorials/start/create.html also maybe?

File: firebase.json (hunk 4 above)
Comment (sebastianbenz) [reply firebase.json]: sgtm

File: firebase.json (hunk 3 above)
Comment (sebastianbenz) [reply firebase.json]: Hm, let's leave as it is then

File: firebase.json (hunk 5 above)
Comment (sebastianbenz) [reply firebase.json]: I was thinking https://amp.dev/documentation/guides-and-tutorials/develop/interactivity/create-interactive.html?format=websites

But guides-tutorials is probably better as well...

Comment (matthiasrohmer) [reply firebase.json]: NVM the exact match is: https://amp.dev/documentation/guides-and-tutorials/learn/combine-amp-pwa.html?format=websites

Comment (matthiasrohmer) [reply firebase.json]: Did you probably slip lines during review? This pair currently is a 1:1 match.

Comment (matthiasrohmer) [reply firebase.json]: Line slip? Exact match would be: https://amp.dev/documentation/guides-and-tutorials/learn/combine-amp-pwa.html?format=websites

Comment (sebastianbenz) [reply firebase.json]: but that doesn't really make sense

Comment (sebastianbenz) [reply firebase.json]: I find PWA confusing when I'm looking for dynamic and interactive...

Comment (sebastianbenz) [reply firebase.json]: Really? The articles in https://www.ampproject.org/docs/media/  are the same as in  https://amp.dev/documentation/guides-and-tutorials/develop/media_iframes_3p

Comment (matthiasrohmer) [reply firebase.json]: But isn't the goal that formerly published links to http://ampproject.org/docs/integration/pwa-amp point to their equivalent on amp.dev to keep context?

File: firebase.json [hunk 6]
```diff
@@ -1629,6 +1629,651 @@
...
         "type": 301
+      },
+      {
+        "source": "/:lang",
+        "destination": "https://amp.dev/:lang/?referrer=ampproject.org",
+        "type": 301
+      },
+      {
+        "source": "/:lang/about/benefits",
+        "destination": "https://amp.dev/:lang/#benefits?referrer=ampproject.org",
```
Comment (sebastianbenz) [inline firebase.json]: hash comes after query

Comment (matthiasrohmer) [reply firebase.json]: Good catch.

**Pull request merged**